src/rust/engine/process_execution/src/lib.rs

```diff
@@ -74,3 +78,31 @@ pub trait CommandRunner: Send + Sync {
 
   fn reset_prefork(&self);
 }
+
+///
+/// A CommandRunner wrapper that limits the number of concurrent requests.
```

Comment:
Forward-looking: I'm not sure how we should handle this for the `remote::CommandRunner` case - the `remote::CommandRunner` has its own threadpool for the grpc environment, which has a fixed size, and having multiple concurrency limits doesn't seem ideal.

Then again, maybe just switching to https://github.com/tower-rs/tower-grpc at some point and using tokio to drive grpc is the answer.

Reply:
Right... I fully expect us to switch to tokio for grpc at some point.